intellij/src/main/kotlin/motif/intellij/hierarchy/ScopeHierarchyBrowser.kt

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2018-2019 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package motif.intellij.hierarchy
+
+import com.intellij.icons.AllIcons
+import com.intellij.ide.hierarchy.HierarchyBrowserBaseEx
+import com.intellij.ide.hierarchy.HierarchyNodeDescriptor
+import com.intellij.ide.hierarchy.HierarchyTreeStructure
+import com.intellij.ide.hierarchy.JavaHierarchyUtil
+import com.intellij.ide.util.treeView.NodeDescriptor
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages.showInfoMessage
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementFactory
+import com.intellij.psi.PsiType
+import com.intellij.psi.presentation.java.ClassPresentationUtil
+import com.intellij.ui.treeStructure.Tree
+import motif.ast.IrType
+import motif.ast.intellij.IntelliJClass
+import motif.ast.intellij.IntelliJType
+import motif.core.ResolvedGraph
+import motif.intellij.MotifProjectComponent
+import motif.intellij.ScopeHierarchyUtils
+import motif.intellij.ScopeHierarchyUtils.Companion.getMotifScopePaths
+import motif.intellij.ScopeHierarchyUtils.Companion.isMotifScopeClass
+import motif.intellij.hierarchy.descriptor.ScopeHierarchyRootDescriptor
+import motif.intellij.hierarchy.descriptor.ScopeHierarchyScopeDescriptor
+import motif.models.Scope
+import java.text.MessageFormat
+import java.util.*
+import javax.swing.JPanel
+import javax.swing.JTree
+import javax.swing.tree.DefaultMutableTreeNode
+
+/*
+ * UI component used to render tree of Motif scopes.
+ */
+class ScopeHierarchyBrowser(
+        val project: Project,
```

Comment:
`private`

Reply:
class is used outside package (from MotifPanel)